gnosis/safe-react
# fix: better network comparison + WC peer tracking

## What it solves
Resolves #2812 + #3579

## How this PR fixes it
The name of the WalletConnect peer is now tracked in the provider middleware alongside the connected wallet when the user is connected to a network. `shouldSwitchNetwork` was improved to prevent this from being over-tracked.

## How to test it
Connect to the Safe via MM as an example and observe the following wallet tracking:
1. Initial connection
2. Account switching
3. Changing chain in the UI then MM

The WC peer should be tracked in the same instances.

The "Switch wallet to {{chainName}}" button should continue to work as before.

## Analytics changes
The WC peer name should now be tracked, tracking as 'Unknown' if one doesn't exist.

## Screenshots
![image](https://user-images.githubusercontent.com/20442784/160093899-9847ea22-50ef-4f96-9209-b7571767dd72.png)
